### DISCUSSION TITLE
shinyjs Quarto compatibility?

### DISCUSSION THREAD

**Discussion by blenback** [daattali/shinyjs#268] [Q&A]
Hi, 
I already created an issue on this at the Quarto repository ([https://github.com/quarto-dev/quarto-cli/issues/5984](url)) but I wanted to implement some shinyjs functionality in a shiny doc created with Quarto and it doesn't appear to be working. I know from the documentation that shinyjs is compatible with interactive markdown docs but does this extend to Quarto?

**Comment by daattali** [comment]
I've never used quarto so I'm honestly not sure, but from my understanding it should be 100% backwards compatible with rmd. If that's true, then it should work.  It sounds like probably a bug in quarto, or maybe it's this https://github.com/rstudio/rmarkdown/issues/2475

**Reply by blenback** [reply]
Thanks, I actually tried the same reprex in a .rmd instead of .qmd and the functionality worked fine so I'm guessing it's a Quarto bug.

**Reply by daattali** [reply]
Could you also try using the latest github version of shinyjs, just for completeness? In the new version, you won't need to specify `rmd=TRUE` anymore, it should be detected automatically. I'm wondering if that works in quarto.

**Reply by blenback** [reply]
Good suggestion, this solved it!

**Reply by daattali** [reply]
That's excellent to know, so the new version works with quarto?

**Reply by blenback** [reply]
Yes, at least for the limited shinjs functions I tried.

**Reply by andrewmacd** [reply]
> Could you also try using the latest github version of shinyjs, just for completeness? In the new version, you won't need to specify `rmd=TRUE` anymore, it should be detected automatically. I'm wondering if that works in quarto.

Just fyi, you still need the rmd=TRUE option in Quarto. The command should probably be updated as Rmarkdown has largely been superseded by Quarto, so the flag should be qmd=TRUE. 

Additionally, it might help to update your website documentation as the useShinyjs() function is required in both the code chunk(s) that define the UI and in the chunk that has the server context code (for reasons I don't totally understand but seems to be necessary).

**Comment by daattali** [comment]
FYI the only difference between specifying `rmd=TRUE` or not is that in rmd mode, the shinyjs scripts are added to the page using `tagList()` whereas in regular shiny apps it's done using `tags$head()`. You can see that here https://github.com/daattali/shinyjs/blob/894346569df4e8198119e9608426d7d60e166aae/R/utils.R#L62C5-L66

That's the only difference. The file extension doesn't matter.
